tests/integration/aic.rs

```diff
@@ -0,0 +1,242 @@
+    let flow_id_new = format!("{}{}", flow_id_orig, "extra");
+    let update_data = json!({
+        "cj_id": cj_event_value_new,
+        "flow_id": flow_id_new,
+    });
+
+    /* CALL */
+    let r = send_put_request(&app, &path, update_data).await;
+    assert_eq!(r.status(), 201);
+    let response: AICResponse = r.json().await.expect("Failed to get JSON response.");
+
+    /* TEST */
+    assert_eq!(aic_orig.id, response.aic_id);
+    // New expires time should be later than the original
+    assert!(response.expires > aic_orig.expires);
+    assert_saved(
+        model,
+        response.aic_id,
+        response.expires.unix_timestamp(),
+        cj_event_value_new,
+        flow_id_new,
+    )
+    .await;
+}
+
+/* Caller sends AIC id, flowId, existing CJEvent value
+    - keep existing AIC id, creation time, expiration time, cjevent value
+    - save new flow ID
+    - return existing expiration time, existing AIC id
+*/
+#[tokio::test]
+async fn aic_update_when_aic_and_cjevent_exists() {
+    /* SETUP */
+    let (app, cj_event_value_orig, flow_id_orig, _data) = setup_aic_test().await;
+    let model = AICModel {
+        db_pool: &app.db_connection(),
+    };
+    let aic_orig = model
+        .create(&cj_event_value_orig, &flow_id_orig)
+        .await
+        .expect("Failed to create test object.");
+    std::thread::sleep(std::time::Duration::from_secs(1));
+    let path = format!("/aic/{}", aic_orig.id);
+
+    let flow_id_new = format!("{}{}", flow_id_orig, "extra");
+    let update_data = json!({
+        "cj_id": cj_event_value_orig,
+        "flow_id": flow_id_new,
+    });
+
+    /* CALL */
+    let r = send_put_request(&app, &path, update_data).await;
+    assert_eq!(r.status(), 201);
+    let response: AICResponse = r.json().await.expect("Failed to get JSON response.");
+
+    /* TEST */
+    assert_eq!(aic_orig.id, response.aic_id);
+    assert_eq!(
+        response.expires.unix_timestamp(),
+        aic_orig.expires.unix_timestamp()
+    );
+    assert_saved(
+        model,
+        response.aic_id,
+        response.expires.unix_timestamp(),
+        cj_event_value_orig,
+        flow_id_new,
+    )
+    .await;
+}
+
+/* Caller sends flowId, CJEvent value, and AIC value but AIC doesn't exist in our DB
+    - create a new AIC id
+    - save creation time, expiration time, AIC id, flow ID, CJ event value
+    - return expiration time, and AIC id
+*/
+#[tokio::test]
+async fn aic_update_when_no_aic_exists() {
+    /* SETUP */
+    let (app, cj_event_value, flow_id, data) = setup_aic_test().await;
+    let model = AICModel {
+        db_pool: &app.db_connection(),
+    };
+
+    /* CALL */
+    let path = format!("/aic/{}", Uuid::new_v4());
+    let r = send_put_request(&app, &path, data).await;
+    assert_eq!(r.status(), 201);
+    let response: AICResponse = r.json().await.expect("Failed to get JSON response.");
+
+    /* TEST */
+    assert_created_response(&response);
+    let saved = assert_saved(
```

Comment:
Maybe not necessary, but would it be useful to have a test that asserts that the uuid you passed in on line 86 is equal to `response.aic_id` ?

Reply:
i'll check

Reply:
Resolved by cleaning up endpoint in fa8ca70